### ISSUE TITLE
help with restroring session

### ISSUE THREAD

**Issue by shivagarwal676** [QupZilla/qupzilla#2243]
my qupzilla crashed and then i restarted it and on the restore session page by mistake i clicked start new session, is it possible to recover that session ?

**Comment by cranes-bill** [comment]
Did you close QupZilla then?
_Sorry for the bad English!_

**Comment by cranes-bill** [comment]
Ok, in general it is possible. Try the following (for Linux):
```
cp -r ~/.config/qupzilla/profiles/default ~/.config/qupzilla/profiles/default.bkp
killall qupzilla
cp ~/.config/qupzilla/profiles/default.bkp/session.dat.old.1 ~/.config/qupzilla/profiles/default/session.dat
```
Then start QupZilla.....

**Comment by shivagarwal676** [comment]
i am using it on a mac

**Comment by cranes-bill** [comment]
No matter what OS you are using. Just find profile directory and look for those files: **session.dat.old session.dat.old.1** _(it depends how many times QupZilla was closed after the crash)_ and follow the similar steps in my previous post.
And again, sorry for my bad English!

**Comment by shivagarwal676** [comment]
Ok, will try thank you very much in advance.

**Comment by shivagarwal676** [comment]
ok so i tried your suggestion but i don't think it worked as i have attached the screenshot from terminal.I don't know anything about coding so can you help me out a bit.
![screen shot 2017-03-09 at 2 02 55 pm](https://cloud.githubusercontent.com/assets/26295335/23742033/4a9e0bac-04d1-11e7-9387-763330985c87.png)

**Comment by cranes-bill** [comment]
I told similar, but not the same (i'm not familiar with Mac directories tree)....
OK, after quick look at the source, i think the profile directory should be in: `/Users/shivagarwal/Library/Application Support/QupZilla`
If it's there, just copy **default** folder on save place.
Start QupZilla, then kill it from the terminal!
From copied folder find file **session.dat.old.1** and copy it to current profile directory _(probably /Users/shivagarwal/Library/Application Support/QupZilla/default_**/session.dat**)
Start QupZilla again. It should ask you to restore the desired session!

**Comment by shivagarwal676** [comment]
thanks for making this simple i understand.but when i copy session.dat.old.1 to the current directory should i rename it to session.dat or just copy it so it will rewrite the dat.old.1 there ?

**Comment by cranes-bill** [comment]
You must rename it (to overwrite **session.dat**)! 
(If do not have session.dat.old.1, copy session.dat.old)

**Comment by shivagarwal676** [comment]
ok

**Comment by srazi** [comment]
Closed using 755ec16598656776e94dbaae9de25f514a592b53

You can open backup sessions by pointing to `File->Sessions->First Backup` or `File->Sessions->Second Backup` and then select `open`
